reconcile/jira_permissions_validator.py

```diff
@@ -109,54 +114,71 @@ def board_is_valid(
-                break
-
-        if not available_states:
-            logging.error(
-                f"[{board.name}] {issue_type} doesn't have any status. Choose a different issue type."
-            )
-            error |= ValidationError.INVALID_ISSUE_TYPE
-
-        reopen_state = board.issue_reopen_state or default_reopen_state
-        if reopen_state.lower() not in [t.lower() for t in available_states]:
-            logging.error(
-                f"[{board.name}] '{reopen_state}' is not a valid state in project. Valid states: {available_states}"
-            )
-            error |= ValidationError.INVALID_ISSUE_STATE
+        if project_issue_type:
+            # Check issue attributes
+            if not project_issue_type.statuses:
+                logging.error(
+                    f"[{board.name}] {issue_type} doesn't have any status. Choose a different issue type."
+                )
+                error |= ValidationError.INVALID_ISSUE_TYPE
 
-        if board.issue_resolve_state and board.issue_resolve_state.lower() not in [
-            t.lower() for t in available_states
-        ]:
-            logging.error(
-                f"[{board.name}] '{board.issue_resolve_state}' is not a valid state in project. Valid states: {available_states}"
-            )
-            error |= ValidationError.INVALID_ISSUE_STATE
+            reopen_state = board.issue_reopen_state or default_reopen_state
+            if reopen_state.lower() not in [
+                t.lower() for t in project_issue_type.statuses
+            ]:
+                logging.error(
+                    f"[{board.name}] '{reopen_state}' is not a valid state in project. Valid states: {project_issue_type.statuses}"
+                )
+                error |= ValidationError.INVALID_ISSUE_STATE
 
-        if board.issue_security_id:
-            security_levels = jira.security_levels()
-            if board.issue_security_id not in [level.id for level in security_levels]:
+            if board.issue_resolve_state and board.issue_resolve_state.lower() not in [
+                t.lower() for t in project_issue_type.statuses
+            ]:
                 logging.error(
-                    f"[{board.name}] {board.issue_security_id} is not a valid security level in project. Valid security ids: "
-                    + ", ".join([
-                        f"{level.name} - {level.id}" for level in jira.security_levels()
-                    ])
+                    f"[{board.name}] '{board.issue_resolve_state}' is not a valid state in project. Valid states: {project_issue_type.statuses}"
                 )
-                error |= ValidationError.INVALID_SECURITY_LEVEL
-        elif board.name in public_projects:
+                error |= ValidationError.INVALID_ISSUE_STATE
+
+            for field in board.issue_fields or []:
+                if not (
+                    project_issue_field := jira.project_issue_field(
```

Comment:
`project_issue_field` is also used outside of `if` section

Reply:
See above